From bug-glibc-bounces+listarch-bug-glibc=sources dot redhat dot com at gnu dot org Tue Aug 05 19:52:26 2003
Return-Path: <bug-glibc-bounces+listarch-bug-glibc=sources dot redhat dot com at gnu dot org>
Delivered-To: listarch-bug-glibc at sources dot redhat dot com
Received: (qmail 31779 invoked from network); 5 Aug 2003 19:52:26 -0000
Received: from unknown (HELO monty-python.gnu.org) (199.232.76.173)
  by sources dot redhat dot com with SMTP; 5 Aug 2003 19:52:26 -0000
Received: from localhost ([127.0.0.1] helo=monty-python.gnu.org)
	by monty-python dot gnu dot org with esmtp (Exim 4 dot 20)
	id 19k7hp-0004mC-J0
	for listarch-bug-glibc at sources dot redhat dot com; Tue, 05 Aug 2003 15:42:49 -0400
Received: from list by monty-python.gnu.org with tmda-scanned (Exim 4.20)
	id 19k7hd-0004lI-No
	for bug-glibc at gnu dot org; Tue, 05 Aug 2003 15:42:37 -0400
Received: from mail by monty-python.gnu.org with spam-scanned (Exim 4.20)
	id 19k7h7-0004TP-8U
	for bug-glibc at gnu dot org; Tue, 05 Aug 2003 15:42:36 -0400
Received: from [32.97.182.102] (helo=e2.ny.us.ibm.com)
	by monty-python dot gnu dot org with esmtp (Exim 4 dot 20) id 19k7h6-0004TH-V4
	for bug-glibc at gnu dot org; Tue, 05 Aug 2003 15:42:05 -0400
Received: from northrelay02.pok.ibm.com (northrelay02.pok.ibm.com
	[9 dot 56 dot 224 dot 150])
	by e2 dot ny dot us dot ibm dot com (8 dot 12 dot 9/8 dot 12 dot 2) with ESMTP id h75Jg3PS037034;
	Tue, 5 Aug 2003 15:42:03 -0400
Received: from D01ML249.pok.ibm.com (d01av02.pok.ibm.com [9.56.224.216])
	by northrelay02 dot pok dot ibm dot com (8 dot 12 dot 9/NCO/VER6 dot 5) with ESMTP id
	h75Jg2vu131828; Tue, 5 Aug 2003 15:42:02 -0400
Importance: Normal
MIME-Version: 1.0
Sensitivity: 
To: schwab at suse dot de, Ulrich Weigand <Ulrich dot Weigand at de dot ibm dot com>
X-Mailer: Lotus Notes Release 5.0.11   July 24, 2002
From: William Jay Huie <wjhuie at us dot ibm dot com>
Message-ID: <OF3D561D81.1A9679A6-ON85256D79.00619378-85256D79.006DAF9F@us.ibm.com>
Date: Tue, 5 Aug 2003 15:41:56 -0400
X-MIMETrack: Serialize by Router on D01ML249/01/M/IBM(Release 6.0.1 [IBM]|June
	10, 2003) at 08/05/2003 15:42:02,
	Serialize complete at 08/05/2003 15:42:02
Cc: bug-glibc at gnu dot org
Subject: fixed building gcc-2.3.2 on s390
X-BeenThere: bug-glibc@gnu.org
X-Mailman-Version: 2.1.2
Precedence: list
List-Id: Bug reports for the GNU standard C library  <bug-glibc.gnu.org>
List-Archive: <http://mail.gnu.org/pipermail/bug-glibc>
List-Post: <mailto:bug-glibc at gnu dot org>
List-Help: <mailto:bug-glibc-request at gnu dot org?subject=help>
List-Subscribe: <http://mail dot gnu dot org/mailman/listinfo/bug-glibc>,
	<mailto:bug-glibc-request at gnu dot org?subject=subscribe>
Content-Type: multipart/mixed; boundary="===============1920858710=="
Sender: bug-glibc-bounces+listarch-bug-glibc=sources dot redhat dot com at gnu dot org
Errors-To: bug-glibc-bounces+listarch-bug-glibc=sources.redhat.com@gnu.org

This is a multipart message in MIME format.
--===============1920858710==
Content-Type: multipart/alternative;
	boundary="=_alternative 006DAF4085256D79_="

This is a multipart message in MIME format.
--=_alternative 006DAF4085256D79_=
Content-Type: text/plain; charset="us-ascii"

The problem was related to a file including an obsolete header file.
        Checking /usr/include/sys/ustat.h shows the following comment;

                /*
                 * This interface is obsolete.  Use <sys/statfs.h> 
instead.
                 */

        Changing the include in the offending file;

osatest2:/usr/src/packages/SOURCES/glibc-2.3.2/sysdeps/unix/sysv/linux # 
diff -u old.ustat.c ustat.c
--- old.ustat.c 2001-07-07 12:21:32.000000000 -0700
+++ ustat.c     2003-08-05 11:24:20.000000000 -0700
@@ -18,7 +18,7 @@
    02111-1307 USA.  */

 #include <errno.h>
-#include <sys/ustat.h>
+#include <sys/statfs.h>
 #include <sys/sysmacros.h>

 #include <sysdep.h>


        Things then build correctly,
        This is on a SLES-8 s390 31bit system using glibc levels 2.2.5-92.
        j

Jay Huie 
  wjhuie@us.ibm.com
  zSeries Linux System Test
  Phone: 845-435-8164

--=_alternative 006DAF4085256D79_=
Content-Type: text/html; charset="us-ascii"


<br><font size=2 face="sans-serif">The problem was related to a file including an obsolete header file.</font>
<br><font size=2 face="sans-serif">&nbsp; &nbsp; &nbsp; &nbsp; Checking /usr/include/sys/ustat.h shows the following comment;</font>
<br>
<br><font size=2 face="sans-serif">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; /*</font>
<br><font size=2 face="sans-serif">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;* This interface is obsolete. &nbsp;Use &lt;sys/statfs.h&gt; instead.</font>
<br><font size=2 face="sans-serif">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;*/</font>
<br>
<br><font size=2 face="sans-serif">&nbsp; &nbsp; &nbsp; &nbsp; Changing the include in the offending file;</font>
<br>
<br><font size=2 face="sans-serif">osatest2:/usr/src/packages/SOURCES/glibc-2.3.2/sysdeps/unix/sysv/linux # diff -u old.ustat.c ustat.c</font>
<br><font size=2 face="sans-serif">--- old.ustat.c 2001-07-07 12:21:32.000000000 -0700</font>
<br><font size=2 face="sans-serif">+++ ustat.c &nbsp; &nbsp; 2003-08-05 11:24:20.000000000 -0700</font>
<br><font size=2 face="sans-serif">@@ -18,7 +18,7 @@</font>
<br><font size=2 face="sans-serif">&nbsp; &nbsp; 02111-1307 USA. &nbsp;*/</font>
<br>
<br><font size=2 face="sans-serif">&nbsp;#include &lt;errno.h&gt;</font>
<br><font size=2 face="sans-serif">-#include &lt;sys/ustat.h&gt;</font>
<br><font size=2 face="sans-serif">+#include &lt;sys/statfs.h&gt;</font>
<br><font size=2 face="sans-serif">&nbsp;#include &lt;sys/sysmacros.h&gt;</font>
<br>
<br><font size=2 face="sans-serif">&nbsp;#include &lt;sysdep.h&gt;<br>
</font>
<br>
<br><font size=2 face="sans-serif">&nbsp; &nbsp; &nbsp; &nbsp; Things then build correctly,</font>
<br><font size=2 face="sans-serif">&nbsp; &nbsp; &nbsp; &nbsp; This is on a SLES-8 s390 31bit system using glibc levels 2.2.5-92.</font>
<br><font size=2 face="sans-serif">&nbsp; &nbsp; &nbsp; &nbsp; j</font>
<br><font size=2 face="sans-serif"><br>
Jay Huie <br>
 &nbsp;wjhuie@us.ibm.com<br>
 &nbsp;zSeries Linux System Test<br>
 &nbsp;Phone: 845-435-8164<br>
</font>
--=_alternative 006DAF4085256D79_=--



--===============1920858710==
Content-Type: text/plain; charset="us-ascii"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Content-Disposition: inline

_______________________________________________
Bug-glibc mailing list
Bug-glibc@gnu.org
http://mail.gnu.org/mailman/listinfo/bug-glibc

--===============1920858710==--


